desktop/main/walletFile.ts

```diff
@@ -276,29 +278,35 @@ export const copyWalletFile = async (filePath: string, outputDir: string) => {
   return newFilePath;
 };
 
-export const listWalletsByPaths = (files: string[]) => {
-  return Promise.all(
+export const loadRawWallets = (files: string[]) =>
+  Promise.all(
     files.map(async (filePath) => {
       try {
         const wallet = await loadRawWallet(filePath);
-        return { path: filePath, meta: wallet.meta };
+        return wallet.meta ? { path: filePath, wallet } : null;
       } catch (err) {
-        return { path: filePath, error: err };
+        return null;
       }
     })
-  ).then(
-    // TODO: Show error to the user?
-    (res) => R.filter(R.has('meta'), res)
+  ).then(R.reject<WalletWithPath | null>(R.isNil)) as Promise<WalletWithPath[]>;
```

Comment:
Such cast may cause problems because TS does not check for props overlapping... for example, you can cast it into `Promise<string[]>` and TypeScript would not warn you.

I guess there might be some point-free solution for correct typing, but I believe more clear and readable will be like this:
```suggestion
  ).then((x: (WalletWithPath | null)[]) => R.reject(R.isNil, x));
```